en_US/data-integration/data-bridge-mqtt.md

```diff
@@ -1,56 +1,71 @@
+<img src="./assets/bridge_egress_rule.png" alt="bridge_egress_rule" style="zoom: 50%;" />
 
-In **ingress** direction, it can be used as the data source of the rule:
+### Connection Pool
 
-<img src="./assets/bridge_igress_rule_link.png" alt="bridge_igress_rule_link" style="zoom:50%;" />
+EMQX allows more than one client to connect to the bridged MQTT broker at the same time. When creating the data bridge, you can set a pool of MQTT client connetions and configure the pool size indicating the number of the client connections in the pool. Enabling connection pool in MQTT data bridge allows for efficient utilization of server resources, leading to greater message throughput and improved concurrency performance. This brings benefits in scenarios with high loads and concurrent connections.
 
-In **egress** direction, it can be used as the action of the rule:
+Because the MQTT protocol requires that a client connects to one broker must has it's own unique client ID, each client in the connection pool is assigned to a unique client ID. To make the client ID predictable, EMQX automatically generates the Client ID following the pattern below:
 
-<img src="./assets/bridge_egress_rule.png" alt="bridge_egress_rule" style="zoom: 50%;" />
+```bash
+[${ClientIDPrefix}:]${BridgeName}:${Mode}:${NodeName}:${N}
+```
 
-In both modes EMQX employs pools of MQTT clients to connect to the bridged brokers, in order to achieve high throughput and make use of available parallelism.
+| Fragment            | Description                                                  |
+| ------------------- | ------------------------------------------------------------ |
+| `${ClientIDPrefix}` | Prefix of the client ID as configured. If not set then whole first fragment is omitted. |
+| `${BridgeName}`     | User-provided name of the bridge.                            |
+| `${Mode}`           | Either `ingress` or `egress`.                                |
+| `${NodeName}`       | [Name of the node](../configuration/cluster.md#node-names) on which the MQTT client is running. |
+| `${N}`              | Number from `1` to the configured size of the connection pool. |
+
+#### Use Connection Pool in Ingress Mode
+
+Although the connection pool applies to both ingress and egress modes, using the connection pool in ingress mode has more requirements. When data bridges in ingress mode are created between MQTT brokers in different MQTT [clusters](../deploy/cluster/introduction.md), clients in the connection pool will receive duplicated messages from the remote broker if they all subscribe to the same topic. In that case, it will bring pressure to the brokers, so it is strongly advised to use [shared subscription](../messaging/mqtt-shared-subscription.md) as a kind of safety measure. For example, you can configure the topic of the remote MQTT Broker to  `$share/name1/topic1` or `$share/name2/topic2/#` if topic filter is used. 
```

Comment:
> requirements

I'd say this is a _caveat_ rather than a _requirement_.

Also, this paragraph kinda lost large part of its initial message. There's no mention anymore that in non-shared subscription case pool downscales to a single client.

> When data bridges in ingress mode are created between MQTT brokers in different MQTT [clusters](../deploy/cluster/introduction.md)

This sounds a bit misleading. Issue arises when user has EMQX cluster with more than 1 node, and configures an ingress MQTT bridge with non-shared topic to subscribe to on the remote broker. The remote broker does not have to be EMQX, it can be anything.

Bottom line is that paragraph was initially about when and how user expectations about _bridging_ will **break** under different configurations (EMQX cluster / single node, shared / non-shared subscription), namely message duplication.

Reply:
Thank you for your comments. I have modified the descriptions.